java/vector/src/test/java/org/apache/arrow/vector/compare/TestRangeEqualsVisitor.java

```diff
@@ -132,6 +136,26 @@ public void testBaseVariableVectorRangeEquals() {
     }
   }
 
+  @Test
+  public void testBaseVariableViewVectorRangeEquals() {
+    try (final ViewVarCharVector vector1 = new ViewVarCharVector("varchar", allocator);
+        final ViewVarCharVector vector2 = new ViewVarCharVector("varchar", allocator)) {
+
+      setVector(vector1, STR1, STR2, STR4, STR3, STR2, STR5, STR1, STR6, STR1, STR2, STR4);
+      setVector(vector2, STR1, STR2, STR4, STR3, STR2, STR5, STR1, STR6, STR1, STR2, STR4);
+
+      RangeEqualsVisitor visitor = new RangeEqualsVisitor(vector1, vector2);
+      // inclusion of long string in the middle
+      assertTrue(visitor.rangeEquals(new Range(1, 1, 3)));
+      // inclusion of long string at the start
+      assertTrue(visitor.rangeEquals(new Range(2, 2, 4)));
+      // inclusion of long string at the end
+      assertTrue(visitor.rangeEquals(new Range(4, 4, 4)));
+      // unequal range
+      assertTrue(visitor.rangeEquals(new Range(8, 0, 3)));
```

Comment:
....they're still equal. Let's test ranges that are not supposed to be equal.

Reply:
ah right! :slightly_smiling_face: I misunderstood.

Reply:
added a few not equal cases for each category except for `inclusion of long string at the end`, since it is not possible to have that case.